src/Trigger.cc

```diff
@@ -373,10 +373,7 @@ void Trigger::Register(Val* val) {
 void Trigger::UnregisterAll() {
     DBG_LOG(DBG_NOTIFIERS, "%s: unregistering all", Name());
 
-    for ( const auto& o : objs ) {
-        notifier::detail::registry.Unregister(o.second, this);
-        Unref(o.first);
```

Comment:
This `Unref()` has been lost? Wouldn't mind sticking with the old variant, given all the comments how the Registry isn't doing reference management.

Reply:
Ah - lost in the switch to the other API. I'll revert (and remove the API, since this shows that it's problematic).

I'm guessing CI ASAN found this via leaks as opposed to memory problems? Alas ASAN on my Mac doesn't do leaks ...

Reply:
> I'm guessing CI ASAN found this via leaks as opposed to memory problems?

Ah yes, those were all LeakSanitizer report.